Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -68,4 +270,51 @@ class TabTrayController: UIViewController, UITabBarDelegate, UITableViewDelegate
         tabManager.removeTab(tab)
         tableView.deleteRowsAtIndexPaths([indexPath], withRowAnimation: UITableViewRowAnimation.Automatic)
     }
-}
\ No newline at end of file
+}
+
+extension TabTrayController : Transitionable {
+    private func getTransitionCell(opts: TransitionOptions, browser: Browser?) -> CustomCell {
+        if let cell = opts["cell"] as? CustomCell {
+            cell.tab = browser
+            return cell
+        } else {
+            let cell = CustomCell(style: UITableViewCellStyle.Default, reuseIdentifier: "id")
+            cell.tab = browser
+            opts["cell"] = cell
+            return cell
+        }
+    }
+
+    func hideTransition(opts: TransitionOptions, vc: UIViewController) {
+        let container = opts["container"] as UIView
+
+        // Create a fake cell that is shown fullscreen
+        if let browser = vc as? BrowserViewController {
```

Comment:
Yuck, we especially don't want to be doing this. Using GeckoApp as a huge sink and referencing it everywhere is what got us into trouble on Android.

But we shouldn't need to, right? We can just use the tabManager directly?